### PR TITLE
Remove build files

### PR DESCRIPTION
**Pull request by JDBetteridge**
This addresses issue #1771 .
I have also bumped the VTK version forward to get the latest wheels available on PyPI.

### CODE REVIEW

**Comment by JDBetteridge** [comment]
I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here

**Comment by wence-** [comment]
> I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here

Is your intention to `--remove-build-files` inside the docker images? That seems like a reasonable thing to aim for. If so, you have not done that.

You can test that the docker build works by changing the `if branch == master` line in the Jenkinsfile (don't forget to remove the lines that push the new images!)

**Comment by JDBetteridge** [comment]
> > I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here
> 
> Is your intention to `--remove-build-files` inside the docker images? That seems like a reasonable thing to aim for. If so, you have not done that.
> 
> You can test that the docker build works by changing the `if branch == master` line in the Jenkinsfile (don't forget to remove the lines that push the new images!)

Ah I had assumed that Docker was using the Jenkins build, when it clearly isn't! Changed now, let's see if it passes.

File: firedrake/paraview_reordering.py [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
 from pyop2.utils import as_tuple
-import importlib
+
 """
-This requires an explentation.
-Vtk has some .so deps that might not be present (e.g. libsm.so (X11 Sessions))
-However, we only need vtkCommonKitPython, which, according to ldd, only cares about
-things that we should expect: libc, libdl.so, libstdc++, libm, libgcc_s.
-Thus, we hackily import the module that lives in vtkCommonKitPython.so
+This requires (even more) explanation.
```
Comment (dham) [inline firedrake/paraview_reordering.py:9]: The reader won't know what "even more" refers to. Just drop it.

File: docker/Dockerfile.vanilla [hunk 2]
```diff
@@ -9,5 +9,5 @@ USER firedrake
 WORKDIR /home/firedrake
 
 # Now install Firedrake.
-RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
-RUN bash -c "python3 firedrake-install --no-package-manager --disable-ssh"
+RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/remove-build-files/scripts/firedrake-install
+RUN bash -c "python3 firedrake-install --no-package-manager --disable-ssh --remove-build-files"
```
Comment (dham) [inline docker/Dockerfile.vanilla:13]: I presume we intend to leave this in here. Having small docker containers sounds like a Good Thing (and updating petsc in a docker container sounds odd).

**Pull request merged**